specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -320,10 +326,20 @@ function useEnabledButtons(tableName: keyof Tables): {
     'disableClone'
   );
   const [disableAdd] = userPreferences.use('form', 'preferences', 'disableAdd');
+
+  const tablesToNotClone = Object.keys(appResourceSubTypes)
+    .filter((key) => appResourceSubTypes[key].name !== undefined)
+    .map((key) => appResourceSubTypes[key]?.name!.toLowerCase());
```

Comment:
since this is not dependent on any variables defined in this scope, could you please move it outside this function (i.e right below it) so that we don't re-compute this on each render?